src/NuGet.Services.AzureSearch/Owners2AzureSearch/Owners2AzureSearchCommand.cs

```diff
@@ -0,0 +1,106 @@
+        private readonly IOwnerSetComparer _ownerSetComparer;
+        private readonly IOwnerIndexActionBuilder _indexActionBuilder;
+        private readonly Func<IBatchPusher> _batchPusherFactory;
+        private readonly IOptionsSnapshot<AzureSearchJobConfiguration> _options;
+        private readonly ILogger<Owners2AzureSearchCommand> _logger;
+
+        public Owners2AzureSearchCommand(
+            IDatabaseOwnerFetcher databaseOwnerFetcher,
+            IOwnerDataClient ownerDataClient,
+            IOwnerSetComparer ownerSetComparer,
+            IOwnerIndexActionBuilder indexActionBuilder,
+            Func<IBatchPusher> batchPusherFactory,
+            IOptionsSnapshot<AzureSearchJobConfiguration> options,
+            ILogger<Owners2AzureSearchCommand> logger)
+        {
+            _databaseOwnerFetcher = databaseOwnerFetcher ?? throw new ArgumentNullException(nameof(databaseOwnerFetcher));
+            _ownerDataClient = ownerDataClient ?? throw new ArgumentNullException(nameof(ownerDataClient));
+            _ownerSetComparer = ownerSetComparer ?? throw new ArgumentNullException(nameof(ownerSetComparer));
+            _indexActionBuilder = indexActionBuilder ?? throw new ArgumentNullException(nameof(indexActionBuilder));
+            _batchPusherFactory = batchPusherFactory ?? throw new ArgumentNullException(nameof(batchPusherFactory));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            if (_options.Value.MaxConcurrentBatches <= 0)
+            {
+                throw new ArgumentException(
+                    $"The {nameof(AzureSearchJobConfiguration.MaxConcurrentBatches)} must be greater than zero.",
+                    nameof(options));
+            }
+        }
+
+        public async Task ExecuteAsync()
+        {
+            _logger.LogInformation("Fetching old owner data from blob storage.");
+            var storageResult = await _ownerDataClient.ReadLatestIndexedAsync();
+
+            _logger.LogInformation("Fetching new owner data from the database.");
+            var databaseResult = await _databaseOwnerFetcher.GetPackageIdToOwnersAsync();
+
+            _logger.LogInformation("Detecting owner changes.");
+            var changes = _ownerSetComparer.Compare(storageResult.Result, databaseResult);
+            var changesBag = new ConcurrentBag<IdAndValue<string[]>>(changes.Select(x => new IdAndValue<string[]>(x.Key, x.Value)));
+            _logger.LogInformation("{Count} package IDs have owner changes.", changesBag.Count);
+
+            if (!changes.Any())
+            {
+                return;
+            }
+
+            _logger.LogInformation(
+                "Starting {Count} workers pushing owners changes to Azure Search.",
+                _options.Value.MaxConcurrentBatches);
+            var workerTasks = Enumerable
```

Comment:
Maybe replace this with https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/Catalog/Helpers/ParallelAsync.cs?

Reply:
Yes, I can do that.